bench/BenchmarkDotNet.Artifacts/results/Polly.Core.Benchmarks.CreationBenchmark-report-github.md

```diff
@@ -0,0 +1,15 @@
+```
+
+BenchmarkDotNet v0.13.6, Windows 11 (10.0.22621.1992/22H2/2022Update/SunValley2) (Hyper-V)
+Intel Xeon Platinum 8370C CPU 2.80GHz, 1 CPU, 16 logical and 8 physical cores
+.NET SDK 7.0.306
+  [Host] : .NET 7.0.9 (7.0.923.32018), X64 RyuJIT AVX2
+
+Job=MediumRun  Toolchain=InProcessEmitToolchain  IterationCount=15  
+LaunchCount=2  WarmupCount=10  
+
+```
+|      Method |       Mean |    Error |   StdDev |   Gen0 | Allocated |
+|------------ |-----------:|---------:|---------:|-------:|----------:|
+| Fallback_V7 |   114.8 ns |  1.84 ns |  2.70 ns | 0.0191 |     480 B |
+| Fallback_V8 | 4,324.7 ns | 21.92 ns | 31.43 ns | 0.2518 |    6504 B |
```

Comment:
The creation in V8 is indeed more expensive as it involves a lot more infra. For this reason the recreation of strategies on hot path should be discouraged.

Reply:
I don't think I have seen a scenario where you are "forced" to recreate a strategy on a hot path. 

At worst you need to fallback to some dynamic value, but that can be archived by passing such value to the fallback action using `ResilienceContext`.

Reply:
In you code the caching could look like:

![image](https://github.com/App-vNext/Polly/assets/103487740/bac16e92-cd53-49d2-b765-0106189227c3)

Reply:
Could you paste the code rather than a picture of it? 😄

Reply:
Sure, I tried to create a draft PR to your repo and got rejected.

``` csharp
public ResilienceStrategy<TResult> GetStrategy<TResult>(ApiEndpointOption endpoint, bool handleExecutionFaults, string resource)
{
    return _registry.GetOrAddStrategy<TResult>($"fallback/{resource}/{handleExecutionFaults}", builder =>
    {
        var shouldHandle = new PredicateBuilder<TResult>()
            .Handle<ApiException>()
            .HandleHttpRequestFault()
            .Handle<TaskCanceledException>();

        if (handleExecutionFaults)
        {
            shouldHandle = shouldHandle
                .Handle<BrokenCircuitException>()
                .Handle<IsolatedCircuitException>()
                .Handle<TimeoutRejectedException>();
        }

        builder
            .AddStrategy(GetStrategy(endpoint, resource))
            .AddFallback(new FallbackStrategyOptions<TResult>()
            {
                FallbackAction = context =>
                {
                    if (context.Context.Properties.TryGetValue(FallbackKeys<TResult>.FallbackValue, out var value) && value != null)
                    {
                        return Outcome.FromResultAsTask(value());
                    }

                    return Outcome.FromResultAsTask<TResult>(default);
                },
                ShouldHandle = shouldHandle,
                StrategyName = $"{endpoint.Name} Fallback",
            });
    });
}
```

Reply:
Do you have the code for `FallbackKeys`?